pbsmetrics/metrics.go

```diff
@@ -8,12 +8,13 @@ import (
 
 // Labels defines the labels that can be attached to the metrics.
 type Labels struct {
-	Source        DemandSource
-	RType         RequestType
-	PubID         string // exchange specific ID, so we cannot compile in values
-	Browser       Browser
-	CookieFlag    CookieFlag
-	RequestStatus RequestStatus
+	Source               DemandSource
+	RType                RequestType
+	PubID                string // exchange specific ID, so we cannot compile in values
+	Browser              Browser
+	CookieFlag           CookieFlag
+	RequestStatus        RequestStatus
+	RequestQueueDuration float64
```

Comment:
Unfortunately, there is no magic (aka reflection) involved with metrics. Adding a field to the labels struct makes the data available to the Influx and Prometheus metric engines, but they would need to specifically look for the new information and forward it in some way to their respective data stores.

Request time is currently tracked by the RecordRequestTime method, taking in both the Labels struct and a duration. That is not appropriate for recording queued request time because RecordRequestTime filters out all non-successful requests. I fear changing that behavior would introduce a breaking change and cause issues with existing monitoring queries folks may have.

I recommending adding a new metrics engine method along the lines of `RecordRequestQueueTime` possibly including the RType/RequestType. You'll need to add new metrics to both Influx (aka go-metrics) and Prometheus.

Reply:
Added initial attempt to support queued request timeout metrics. Unit tests will be added. Please review if this makes sense to you and I move in right direction.